packages/yourloops/test/lib/auth/hook.test.tsx

```diff
@@ -320,4 +320,35 @@ describe('Auth hook', () => {
       expect(auth.getFlagPatients()).toEqual([userId])
     })
   })
+
+  describe('completeSignup', () => {
+    it('should update user profile, preferences and settings', async () => {
+      jest.spyOn(UserApi, 'getProfile').mockResolvedValueOnce(undefined)
+      jest.spyOn(UserApi, 'getPreferences').mockResolvedValueOnce(undefined)
+      jest.spyOn(UserApi, 'getSettings').mockResolvedValueOnce(undefined)
+      jest.spyOn(UserApi, 'updateProfile').mockResolvedValue(undefined)
+      jest.spyOn(UserApi, 'updateSettings').mockResolvedValue(undefined)
+      jest.spyOn(UserApi, 'updatePreferences').mockResolvedValue(undefined)
+      const signupForm = {
+        profileFirstname: 'Tim',
+        profileLastname: 'Hagine',
+        preferencesLanguage: 'fr',
+        profileCountry: 'fr'
+      } as SignupForm
+      await initAuthContext()
+      console.log(auth.user)
+
+      expect(auth.user.profile).toBeUndefined()
+      expect(auth.user.preferences).toBeUndefined()
+      expect(auth.user.settings).toBeUndefined()
+
+      await act(async () => {
+        await auth.completeSignup(signupForm)
+      })
+
+      expect(auth.user.profile.fullName).toEqual('Tim Hagine')
```

Comment:
Why test only `auth.user.profile.fullName`? There are other fields to test in `auth.user.profile.`! Do not stop so close from perfection!